src/test/java/edu/ucla/library/services/util/props/PropsWriterTest.java

```diff
@@ -0,0 +1,132 @@
+
+package edu.ucla.library.services.util.props;
+
+import static com.github.stefanbirkner.systemlambda.SystemLambda.catchSystemExit;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.junit.contrib.java.lang.system.SystemErrRule;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+
+/*
+ * Unit tests for ProsWriter.java
+ */
+public class PropsWriterTest {
+
+  /**
+   * Default properties file.
+  */
+    private static final String DEFAULTS =
+        "src/test/resources/properties.default";
+
+  /**
+   * Output template file.
+  */
+    private static final String TEMPLATE =
+        "src/test/resources/properties.tmpl";
+
+  /**
+   * Final output file.
+  */
+    private static final String OUTPUT =
+        "src/test/resources/output.properties";
+
+  /**
+   * Fake file name used to cause app failures.
+  */
+    private static final String FAKE =
+        "src/test/resources/not.there";
+
+  /**
+   * Unreadable file
+  */
+    private static final String LOCKED =
+        "src/test/resources/properties.locked";
+
+  /**
+   * rule that allows catching System.err output in tests.
+  */
+    @Rule
+    public final SystemErrRule mySystemErrRule = new SystemErrRule().enableLog();
+
+  /**
+   * Cleans up after a PropsWriter test.
+   *
+   * @throws IOException If there is a problem deleting the file
+  */
+    @After
+    public void tearDown() throws IOException {
+        // delete the output file if left over from test run
+        Files.deleteIfExists(FileSystems.getDefault().getPath(OUTPUT));
+    }
+
+  /**
+   * Tests happy path for app execution.
+   *
+  */
+    @Test
+    public void testBuildProperties() {
+        PropsWriter.buildProperties(DEFAULTS, TEMPLATE, OUTPUT);
+        assertTrue(Files.exists(FileSystems.getDefault().getPath(OUTPUT)));
+    }
+
+  /**
+   * Tests that app catches nonexistent files.
+   *
+  */
+    @Test
+    public void testCheckExists() throws Exception {
+        final int statusCode = catchSystemExit(() -> {
+            PropsWriter.checkExists(FAKE);
+        });
+        assertTrue(mySystemErrRule.getLog().contains("File must exist"));
+        assertEquals(ExitCodes.DOESNT_EXIST_ERROR, statusCode);
+    }
+
+  /**
+   * Tests that app catches nonreadable files.
+   *
+  */
+    @Test
+    public void testCheckReadable() throws Exception {
+        final int statusCode = catchSystemExit(() -> {
+            PropsWriter.checkReadable(FAKE);
```

Comment:
Do you mean to use the LOCKED file defined above for this one? It's not currently being used, but I think this is the one you were going to set as unreadable?

Reply:
Defined new file without read permissions (NOREAD) and use that for `testCheckReadable()`